src/components/ListItem.jsx

```diff
@@ -5,30 +5,27 @@ import { useEffect, useState } from 'react';
 export function ListItem({ name, listToken, itemId, data }) {
 	const [checkedState, setCheckedState] = useState(data.checked);
 
-	const handleChange = (event) => {
-		// console.log(event.target.checked)
-		if (event.target.checked) {
-			setCheckedState(true);
-		} else {
-			setCheckedState(false);
-		}
-		updateItem(listToken, itemId, checkedState);
+	const handleChange = () => {
+		setCheckedState(!data.checked);
+		updateItem(listToken, itemId, !data.checked);
```

Comment:
Instead of having a if/else conditional, we can just set the checkedstate to the opposite of what is in the database. And once that's done, we can update the item with the opposite of what the checked box is.

I think the issue is that the setCheckedState only executes after the function is done.